### PR TITLE
Avoid map shift when scrolling on iOS

### PR DESCRIPTION
**Pull request by jmagman**
Avoid the visible region of the Map wiggling around during scrolling by setting `kGMSMapViewPaddingAdjustmentBehaviorNever` as suggested by @erdnj in https://github.com/flutter/flutter/issues/107913#issuecomment-1190452082

Update the example app to allow the **Map coordinates** page to be scrollable, which reproduces the issue before this PR.  Change the visible region UI from a button to a constantly-updating text label.

Before:
https://user-images.githubusercontent.com/682784/180588745-74069b72-ce64-450e-b3ad-38fc23e8ec0c.mov

After:
https://user-images.githubusercontent.com/682784/180588750-fa54b2d5-43ab-4d77-9725-1e5f6b0da162.mov

Add a native test to validate.  I tried to reproduce this from the dart tests instead of the example app 
https://github.com/flutter/plugins/blob/0cd07eaeaa06f7d5322e34ce1f55523cd732cb11/packages/google_maps_flutter/google_maps_flutter/example/integration_test/google_maps_test.dart#L480
but it never reproduced while pumping the widgets or frames or anything else I could think of.

Fixes https://github.com/flutter/flutter/issues/107913

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter/example/ios/Podfile [hunk 1]
```diff
@@ -34,6 +34,9 @@ target 'Runner' do
 
     pod 'OCMock', '~> 3.9.1'
   end
+  target 'RunnerUITests' do
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/example/ios/Podfile:37]: I wanted to use `CLLocationCoordinate2D` in these tests so I changed this to allow importing `GoogleMaps`.

File: packages/google_maps_flutter/google_maps_flutter/example/ios/Runner.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -10,6 +10,7 @@
 		1498D2341E8E89220040F4C2 /* GeneratedPluginRegistrant.m in Sources */ = {isa = PBXBuildFile; fileRef = 1498D2331E8E89220040F4C2 /* GeneratedPluginRegistrant.m */; };
 		3B3967161E833CAA004F5970 /* AppFrameworkInfo.plist in Resources */ = {isa = PBXBuildFile; fileRef = 3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */; };
 		4510D964F3B1259FEDD3ABA6 /* libPods-Runner.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 7755F8F4BABC3D6A0BD4048B /* libPods-Runner.a */; };
+		4A097997B7B27CE82FFC3AB8 /* libPods-RunnerUITests.a in Frameworks */ = {isa = PBXBuildFile; fileRef = DC8ED0578E8D540BBDA17645 /* libPods-RunnerUITests.a */; };
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/example/ios/Runner.xcodeproj/project.pbxproj:13]: All the changes in the Xcode project are due to the Podfile `RunnerUITests` search path change.

File: packages/google_maps_flutter/google_maps_flutter/example/ios/RunnerUITests/GoogleMapsUITests.m [hunk 3]
```diff
@@ -81,12 +82,71 @@ - (void)testMapCoordinatesPage {
...
+  [scan scanString:@"VisibleRegion:\nnortheast: LatLng(" intoString:NULL];
+  double northeastLatitude;
+  [scan scanDouble:&northeastLatitude];
+  [scan scanString:@", " intoString:NULL];
+  XCTAssertNotEqual(northeastLatitude, 0);
+  double northeastLongitude;
+  [scan scanDouble:&northeastLongitude];
+  XCTAssertNotEqual(northeastLongitude, 0);
+
+  [scan scanString:@"),\nsouthwest: LatLng(" intoString:NULL];
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/example/ios/RunnerUITests/GoogleMapsUITests.m:140]: Not the most scientific thing I've ever written but hey, it's a test.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Review by cyanglaz — APPROVED** [review]
LGTM!

**Pull request merged**